run-build-functions.sh

```diff
@@ -697,6 +698,7 @@ cache_artifacts() {
   cache_cwd_directory ".venv" "python virtualenv"
   cache_cwd_directory ".build" "swift build"
   cache_cwd_directory ".netlify/plugins" "build plugins"
+  cache_cwd_directory ".netlify/rust-functions-cache" "Rust functions"
```

Comment:
I am not sure whether there is a good reason for it, but it seems like every cached directory is created with `mkdir -p` at the top of the file.

https://github.com/netlify/build-image/blob/653805ca4a64301556e56dc4b321ef8fc20cbb7c/run-build-functions.sh#L53

I am wondering if this should be the case here as well?

Reply:
I wondered about that myself. I left it out because I wanted to make sure we only print log messages about a Rust functions cache if people are indeed using Rust functions, otherwise they might feel like their builds are taking longer due to a feature they're not even using.

I believe the current logic will work because:

- `cache_cwd_directory` uses `move_cache` (https://github.com/netlify/build-image/blob/xenial/run-build-functions.sh#L847)
- `move_cache` checks whether the directory exists — if it does, it prints a log message and copies the directory (https://github.com/netlify/build-image/blob/xenial/run-build-functions.sh#L765)

I think the only risk of not using `mkdir -p` is if we try to copy file files to `$PWD/.netlify/rust-functions-cache` and `$PWD/.netlify` doesn't exist. At the moment, we know that it's being created [as part of the build plugins directory](https://github.com/netlify/build-image/pull/602/files#diff-6ad81dc23c4c92a6e6193db3b05e306293aa75ee25d48ae952c6259c601e8b23R54-R56), but I've added a comment to make it clear that it's also required for the Rust functions cache.

Does that make sense, or am I missing some detail here?

Reply:
Makes sense! :+1: